qubes/events.py

```diff
@@ -157,50 +160,92 @@ def _fire_event_in_order(self, order, event, kwargs):
                     key=(lambda handler: hasattr(handler, 'ha_bound')),
                     reverse=True):
                 effect = func(self, event, **kwargs)
+                # "is True" to rule out unittest.mock.Mock...
+                if getattr(func, 'ha_async', False) is True:
```

Comment:
Now I think, couldn't we replace this one with `asyncio.iscoroutinefunction(func)` (not to be confused with `inspect.iscoroutinefunction()`, which is `False` on `@asyncio.coroutine`s)?

That way we'd deal with one corner case:
```python
class MyClass(qubes.events.Emitter):
  @qubes.events.handler('my-event', is_async=True)
  # NOT asyncio.coroutine
  def on_my_event(self, event):
    return asyncio.Future()
```

With `iscoroutinefunction`, the above example, admittedly valid but unreadable, becomes forbidden. And we wouldn't have to remember about `is_async=` and keep it in sync with reality.

Argument against: there are 2 incompatible versions of `iscoroutinefunction()` in python's stdlib (the difference is around `@asyncio.coroutine` as above) and 2.5 versions of `iscoroutine()` (there is also `isinstance(obj, types.CoroutineType)`,which is same as `inspect.iscoroutine()` and not compatible with `asyncio.iscoroutine()`). We can get it right, but after that, we are in the area of "what could possibly go wrong".